core/scheduler/service/replay_worker.go

```diff
@@ -266,20 +275,21 @@ func (w *ReplayWorker) fetchRuns(ctx context.Context, replayReq *scheduler.Repla
 	return w.scheduler.GetJobRuns(ctx, replayReq.Replay.Tenant(), jobRunCriteria, jobCron)
 }
 
-func (w *ReplayWorker) CancelReplayRunsOnScheduler(ctx context.Context, replay *scheduler.Replay, jobCron *cron.ScheduleSpec, runs []*scheduler.JobRunStatus) []*scheduler.JobRunStatus {
+func (w *ReplayWorker) CancelReplayRunsOnScheduler(ctx context.Context, replay *scheduler.Replay, jobCron *cron.ScheduleSpec, runs []*scheduler.JobRunWithDetails) []*scheduler.JobRunStatus {
 	var canceledRuns []*scheduler.JobRunStatus
 	for _, run := range runs {
-		logicalTime := run.GetLogicalTime(jobCron)
+		runState := scheduler.JobRunStatus{
+			ScheduledAt: run.ScheduledAt,
+			State:       run.State,
```

Comment:
Let's verify whether the canceled runs will still have the origin state or the updated ones. I believe we are expecting `canceled` status?

Reply:
yeah you are right this should be   scheduler.StateCanceled